pallets/asset-index/src/lib.rs

```diff
@@ -869,8 +873,26 @@ pub mod pallet {
 
 			// mint SAFT into the treasury's account
 			T::Currency::deposit(asset_id, &Self::treasury_account(), units)?;
+
+			// convert nav to pint
+			let pint = if Self::index_token_issuance().into() == 0_u128 || Self::saft_nav()? == FixedU128::zero() {
+				nav
+			} else {
+				(if Self::net_saft_value(asset_id).into() == 0_u128 {
+					Self::calculate_nav_proportion_with_value(nav, Self::saft_nav()?)?
+				} else {
+					Self::saft_asset_proportion(asset_id)?
+				})
+				.reciprocal()
+				.ok_or(Error::<T>::AssetUnitsOverflow)?
+				.checked_mul_int::<u128>(nav.into())
+				.ok_or(Error::<T>::AssetUnitsOverflow)?
+				.try_into()
+				.map_err(|_| Error::<T>::AssetUnitsOverflow)?
+			};
```

Comment:
These lines are for type conversion, we can not `Price::checked_from_rational(nav, Self::nav()?).ok_or(ArithmeticError::Overflow)?` directory since `Self::nav()` returns `FixedU128` which doesn't have the trait `FixedOpera` FixedPointOperand